### PR TITLE
[feature](log)check and log holding lock time when it exceeds threshold

### PR DESCRIPTION
**Pull request by hust-hhb**
Sometimes the competition of lock is fierce in DatabaseTransactionMgr, which may lead to publish time out, i think we should have a log to hint these lock competition.
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java [hunk 1]
```diff
@@ -141,20 +143,30 @@ public class DatabaseTransactionMgr {
...
     // not realtime usedQuota value to make a fast check for database data quota
     private volatile long usedQuotaDataBytes = -1;
+    private long lockWriteStart;
+
+    private long lockReportingThresholdMs = Config.lock_reporting_threshold_ms;
 
     protected void readLock() {
+        long waitReadStart = System.currentTimeMillis();
         this.transactionLock.readLock().lock();
+        checkAndLogWaitTime(waitReadStart, System.currentTimeMillis());
```
Comment (zhannngchen) [inline fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java:153]: It's not necessary to log readlock wait time, long wait time is always caused by some thread holding writelock too long, and further more, if some thread hold's writelock for a long time, after it unlocked, there's lots of threads get the readlock, they all print such log, especially with stacktrace, which is quite slow.

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 2]
```diff
@@ -2070,5 +2070,11 @@ public class Config extends ConfigBase {
      */
     @ConfField
     public static String bdbje_file_logging_level = "ALL";
+
+    /**
+     * When the acquire/held lock time exceeds the threshold, need to report it.
+     */
+    @ConfField
+    public static long lock_reporting_threshold_ms = 500l;
```
Comment (zhannngchen) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2078]: It's too large, I think 1000ms is big enough

File: fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java (hunk 1 above)
Comment (hust-hhb) [reply fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java]: Thanks for comment, i have changed the code.
I agree to remove lock wait time, and i thinks it's only necessary to record the time of holding write lock, which is sufficient to reflect to situation of lock.

**Comment by hust-hhb** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.89 seconds
 stream load tsv:          467 seconds loaded 74807831229 Bytes, about 152 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          76 seconds loaded 1101869774 Bytes, about 13 MB/s
 stream load parquet:          33 seconds loaded 861443392 Bytes, about 24 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230323011234_clickbench_pr_119032.html

**Review by zhannngchen — APPROVED** [review]

**Comment by zhannngchen** [comment]
run buildall

**Pull request merged**